CMakeLists.txt

```diff
@@ -262,7 +262,10 @@ if(SDL12DEVEL)
       set(SDL_CFLAGS "-D_GNU_SOURCE=1 -D_REENTRANT")
       set(SDL_RLD_FLAGS "")  # !!! FIXME: this forces rpath, which we might want?
       set(SDL_LIBS "-lSDL")
-      set(SDL_STATIC_LIBS ${CMAKE_DL_LIBS})
+      set(SDL_STATIC_LIBS "")
+      foreach(lib ${CMAKE_DL_LIBS})
+          set(SDL_STATIC_LIBS "-l${lib}")
```

Comment:
This should check whether `${CMAKE_DL_LIBS})` is an empty string
which cmake can very well do.

Reply:
It does not. `foreach` ignores empty values:
```cmake
cmake_minimum_required(VERSION 3.0)
project(a C)
set(A "1;;;2;;;;3")
foreach(a ${A})
  message("a ${a}")
endforeach()
```
prints
```
a 1
a 2
a 3
```

This is probably done because a lot of cmake code does: `list(SOME_VALUE APPEND ${SOME_OPTIONAL_LIBRARY})`.

Reply:
Or, does `foreach` handles that by itself??

Reply:
To be honest, it is [not documented](https://cmake.org/cmake/help/latest/command/foreach.html).
So i'm not 100% sure about whether this is documented behavior.

Reply:
OK, will you keep it as is or you want to make changes before merging?

Reply:
> OK, will you keep it as is or you want to make changes before merging?

No changes needed? Shall I merge?

Reply:
Okay by me!

Reply:
OK then.